src/java.base/share/classes/sun/security/util/CryptoAlgorithmConstraints.java

```diff
@@ -63,13 +64,20 @@ public static boolean permits(String service, String algo) {
             new SoftReference<>(null);
 
     /**
-     * Initialize algorithm constraints with the specified security property.
+     * Initialize algorithm constraints with the specified security property
+     * {@code propertyName}. Note that if a system property of the same name
+     * is set, it overrides the security property.
      *
      * @param propertyName the security property name that define the disabled
      *        algorithm constraints
      */
     CryptoAlgorithmConstraints(String propertyName) {
         super(null);
+        String val = System.getProperty(propertyName);
+        // Override the security property with system property value if set
+        if (val != null) {
+            Security.setProperty(propertyName, val);
+        }
```

Comment:
I don't think you should set the security property with the value of the system property. This should be:

```
if (val == null) {
    val = Security.getProperty(propertyName);
}
```

Or you can just call `sun.security.util.SecurityProperties.getOverridableProperty()`.

Reply:
The property value retrieval is inside the parent class. Yes, it's a little like a hack. Let me modify the parent class to achieve this more gracefully then.